tests/trainer/test_trainer.py

```diff
@@ -2095,130 +2089,114 @@ def training_step(self, batch, batch_idx):
     [
         (
             dict(strategy=None, gpus=None),
-            dict(_strategy_type=None, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=1),
+            dict(_strategy_type="single_device", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy="dp", gpus=None),
-            dict(_strategy_type=None, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=1),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy="ddp", gpus=None),
-            dict(_strategy_type=None, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=1),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy="ddp", num_processes=2, gpus=None),
-            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=2),
+            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.CPU, num_gpus=0),
```

Comment:
Agreed.

Reply:
Yeah, the "ddp" == " _StrategyType.DDP" 
I haven't update everything because I want reviewer to notice which one has actually changed. 

Also, I think we can have a seperate PR to clean up '_strategy_type' as it's everywhere..

Reply:
@awaelchli @kaushikb11 all updated to str and deleted `_strategy_type `